plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -971,7 +971,7 @@ def _update_location(package_element, filename):
         :type  filename:        basestring
         """
         location_element = package_element.find('location')
-        location_element.set('href', filename)
+        location_element.set('href', file_utils.make_packages_relative_path(filename))
 
     def get_symlink_name(self):
```

Comment:
This method should return file_utils.make_packages_relative_path(self.filename) so that the rsync distributor can also publish using the same directory structure.